### ISSUE TITLE
Add CHANGELOG

### ISSUE THREAD

**Issue by kastman** [psychopy/psychojs#64]
Hi @apitiot and @dvbridges ,

I'm trying to keep track of all the awesome and wonderful new things you all are doing with the rapid iteration of psychojs, but aside looking through each PR individually there isn't an easy way: there's no CHANGELOG, and the tags aren't being pushed to the github remote so I can't  look at all the code changes (e.g. compare 3.1.4 -> 3.1.5 in github's viewer or git diff revisions). 

For example, did you ever add the snapshots that you were going to talk about about in #49 ? I don't see it referenced in another issue after checking quick, but I likely missed it. :) 

I'm happy to try to add in a few things, but getting lost in the updates and not having any way to track them is a barrier to entry, and one that should be pretty low effort. Thanks for considering the request,

**Comment by peircej** [comment]
Yes, admittedly our dev model for PsychoJS so far has focussed on Alain getting the code written, and doing so fast enough that we weren't really ready to be using the standard open-source contribution model. That is, we've been making the code available but, yes, it hasn't been easy for people to contribute yet.

We're just coming to the point I think where that is changing. 

For versioning we are now organising that here (rather than merely on Pavlovia) but using branches instead of tags. The issue is that, whereas for Python a version becomes fixed on release, for JS that isn't wise. In Python, once a file is downloaded to a user's computer it's obviously fixed. But for JS, the file remains on our servers (Pavlovia) and we can/should still apply critical bug-fixes after release because the files are being loaded live on each use. So a release is a relatively short branch that might have fixes applied but no further features added (a bit like PsychoPy's release series 3.1.x). That might change when things have stabilised further - we might be able to move to a more traditional dev model using tags.

Furthermore, we've got something of an issue to solve around version number because a) we want them to match up with those for PsychoPy (because a version of PsychoPy has to output code for the appropriate version of PsychoJS) but b) the version number may not carry the same semantic meaning for PsychoJS because we might add a major new feature to PsychoPy but not JS and vice-versa.

Basically, it's complicated!

But having a changelog to keep track of what has happened would still be appropriate, obvs! ;-)

**Comment by apitiot** [comment]
Further to @kastman 's excellent suggestion, I have just added a change log and will make sure to keep it absolutely up to date.

The GitHub repository should also now reflect the state of development with the Master branch corresponding to the next release milestone, and separate branches corresponding to PsychoPy's releases, e.g. 3.1.4, 3.1.5, etc.

Alain
